test/unit/Fixture/PHP85ClassForSourceStubber.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Roave\BetterReflectionTest\Fixture;
+
+use JsonSerializable;
+use Serializable;
+
+class ParentClassForSourceStubber
+{
+    public function methodFromParentClass()
+    {
+    }
+}
+
+interface ImplementedInterfaceForSourceStubber extends JsonSerializable
+{
+    public function methodFromInterface();
+}
+
+trait UsedParentTraitForSourceStubber
+{
+    public $propertyFromParentTrait;
+
+    public function methodFromParentTrait()
+    {
+    }
+}
+
+trait UsedTraitForSourceStubber
+{
+    use UsedParentTraitForSourceStubber;
+
+    protected $propertyFromTrait;
+
+    public function methodFromTrait()
+    {
+    }
+}
+
+trait UsedTraitToAliasForSourceStubber
+{
+    public function methodFromTraitToAlias()
+    {
+    }
+}
+
+/**
+ * Class comment
+ */
+abstract class PHP85ClassForSourceStubber extends ParentClassForSourceStubber implements ImplementedInterfaceForSourceStubber, Serializable
```

Comment:
This class is filled to the brim with small edge cases: are there specific sections from PHP 8.5 that were relevant, or is this an "upgrade of the stubbing with some additions"?